tests/features/utils/data.py

```diff
@@ -73,4 +86,46 @@ def create_test_document_reference(items: dict) -> DocumentReference:
             )
         ]
 
+    if items.get("supercedes"):
+        base_doc_ref.relatesTo = [
+            DocumentReferenceRelatesTo(
+                code="replaces",
+                target=Reference(
+                    type="DocumentReference",
+                    identifier=Identifier(
+                        system="https://fhir.nhs.uk/Id/ods-organization-code",
+                        value=items["supercedes"],
+                    ),
+                ),
+            )
+        ]
+
     return base_doc_ref
+
+
+def create_test_document_reference_with_defaults(
+    section: str, custom_data: str, pointer_id: str = "TSTCUS-sample-id-00000"
+) -> str:
+    """
+    Builds a DocumentReference request body for testing purposes,
+    using a valid example DocumentReference body as a default, but
+    with one top-level section of the resource overwritten by a json string
+    """
+
+    doc_ref_text = f"""{{ "resourceType": "DocumentReference",
+      "id": "{pointer_id}",
+      "status": "current",
+      "docStatus": "final",
+      {custom_data if section=="masterIdentifier" else DEFAULT_TEST_MASTER_ID},
+      {custom_data if section=="subject" else DEFAULT_TEST_SUBJECT},
+      {custom_data if section=="custodian" else DEFAULT_TEST_CUSTODIAN},
+      {custom_data if section=="author" else DEFAULT_TEST_AUTHOR},
+      {custom_data if section=="type" else DEFAULT_TEST_TYPE},
+      {custom_data if section=="category" else DEFAULT_TEST_CATEGORY},
+      {custom_data if section=="description" else DEFAULT_TEST_DESCRIPTION},
+      {custom_data if section=="security" else DEFAULT_TEST_SECURITY_LABEL},
+      {custom_data if section=="content" else DEFAULT_TEST_CONTENT},
+      {custom_data if section=="context" else DEFAULT_TEST_CONTEXT}
+    }}"""
```

Comment:
I thought about that first but then I wanted something where the whole document was just a string because failure scenario tests were requested where for example, there are duplicate keys in the request body, and my understanding was that a dictionary would just replace the key, or is there a way around that?